js/guide.js

```diff
@@ -0,0 +1,117 @@
+// JavaScript for Tour Guide
+
+const steps = [
+    { id: "level-description", text: "Welcome to the challenge! This is the task you need to complete." },
+    { id: "level-selection", text: "Check out the current levels and navigate to others to progress." },
+    { id: "hint-icon", text: "Use the hint label if you get stuck on any level." },
+    { id: "target-element", text: "This is the target element you need to create." },
+    { id: "css-container", text: "Write any CSS properties in the code editor to style the target element." },
+    { id: "run-code", text: "Click the 'Run' button to see if you've achieved your target." },
+    { id: "styled-element", text: "Here is the styled element you've crafted!" },
+    { id: "score-tracker", text: "Keep an eye on your score to track your progress." }
+];
+
+let currentStepIndex = 0;
+const tooltip = document.getElementById('tourTooltip');
+const tooltipText = document.getElementById('tooltipText');
+
+function initializeTour() {
+    document.getElementById('startTour').onclick = startTour;
+    document.getElementById('nextStep').onclick = nextStep;
+    document.getElementById('prevStep').onclick = prevStep;
+    document.getElementById('skipTour').onclick = endTour;
+}
+
+function startTour() {
+    //currentStepIndex = 0;
+    showStep();
+}
+
+function nextStep() {
+    if (currentStepIndex < steps.length - 1) {
+        endTour();
```

Comment:
check if this is needed , if not remove it , if it does not help, then there is no need for it